app/src/main/kotlin/compiler/intermediate_form/ControlFlowGraph.kt

```diff
@@ -11,8 +11,14 @@ data class ControlFlowGraph(
     val conditionalTrueLinks: ReferenceMap<IFTNode, IFTNode>,
     val conditionalFalseLinks: ReferenceMap<IFTNode, IFTNode>
 ) {
-    val finalTreeRoots: List<IFTNode> get() = treeRoots.filter {
-        it !in unconditionalLinks && it !in conditionalTrueLinks && it !in conditionalFalseLinks
+    val finalTreeRoots: List<Pair<IFTNode, CFGLinkType>> get() = treeRoots.filter {
```

Comment:
Can we create a class e.g. `OutLink`, with `IFTNode` and `CFGLinkType`?

Reply:
This type (`List<Pair<IFTNode, CFGLinkType>>`) is also used by control flow generators, so this change would make the property inconvenient to use.

Reply:
Yes, but in control flow generators also List<OutLink> would look better. But perhaps too much effort now, let's leave it.